src/main/java/com/my/FoodTruckBackend/customer/CustomerRepository.java

```diff
@@ -0,0 +1,49 @@
+package com.my.FoodTruckBackend.customer;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.http.HttpStatus;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.server.ResponseStatusException;
+
+@Repository
+@RequiredArgsConstructor
+@Slf4j
+public class CustomerRepository {
+    private final JdbcTemplate jdbcTemplate;
+
+    public List<CustomerDto> getListOfCustomers() {
+        String sql = "SELECT * FROM customer";
+
+        return jdbcTemplate.query(sql, new BeanPropertyRowMapper<>(CustomerDto.class));
+    }
+
+    public CustomerDto createNewCustomers(NewCustomerRequestBody newCustomerRequestBody) {
+        String sql = "INSERT INTO customer (first_name, last_name, email, password) VALUES (?, ?, ?, ?) RETURNING *";
+
+        try {
+            Customer newCustomer = jdbcTemplate.queryForObject(
+                sql,
+                new BeanPropertyRowMapper<>(Customer.class),
+                newCustomerRequestBody.getFirstName(),
+                newCustomerRequestBody.getLastName(),
+                newCustomerRequestBody.getEmail(),
+                newCustomerRequestBody.getPassword()
+            );
+            return new CustomerDto(
+                newCustomer.getId(),
+                newCustomerRequestBody.getFirstName(),
+                newCustomerRequestBody.getLastName(),
+                newCustomerRequestBody.getEmail()
+            );
```

Comment:
this works, but you don't need to manually map the customer to your `CustomerDto` yourself. Let the beanWrapper do it

```java
return jdbcTemplate.queryForObject(
                sql,
                new BeanPropertyRowMapper<>(CustomerDto.class),
                newCustomerRequestBody.getFirstName(),
                newCustomerRequestBody.getLastName(),
                newCustomerRequestBody.getEmail(),
                newCustomerRequestBody.getPassword()
            );
```

Reply:
This has been changed

Reply:
Understood, this has been altered